### PR TITLE
[RDY] ci: Fold output in travis web interface

### PR DESCRIPTION
**Pull request by ZyX-I**
Should make Web interface nicer to use.

### CODE REVIEW

File: ci/common/suite.sh [hunk 1]
```diff
@@ -11,14 +11,37 @@ FAIL_SUMMARY=""
...
+  NVIM_TEST_SAVED_OPTS=
 }
 
 exit_suite() {
+  NVIM_TEST_SAVED_OPTS="$-"
+  # Do not allow unneeded trace outside of a fold
+  set +x
+  travis_fold end "${NVIM_TEST_CURRENT_SUITE}"
   if test $FAILED -ne 0 ; then
     echo "Suite ${NVIM_TEST_CURRENT_SUITE} failed, summary:"
```
Comment (ZyX-I) [inline ci/common/suite.sh]: Summary intentionally out of the fold

**Comment by justinmk** [comment]
To see the tests one expands the `+NVIM_TEST_SAVED_OPTS=` section, then the `++set -` sub-section. Oh, but I see there are nice descriptive headers in the right column, e.g. `-tests-functionaltests`.

File: ci/common/suite.sh [hunk 2]
```diff
@@ -11,14 +11,37 @@ FAIL_SUMMARY=""
 END_MARKER="$BUILD_DIR/.tests_finished"
 FAIL_SUMMARY_FILE="$BUILD_DIR/.test_errors"
 
+ANSI_CLEAR="\033[0K"
+
+travis_fold() {
+  local action="$1"
+  local name="$2"
+  name="$(echo -n "$name" | tr '\n\0' '--' | sed -r 's/[^A-Za-z0-9]+/-/g')"
+  name="$(echo -n "$name" | sed -r 's/-$//')"
```
Comment (justinmk) [inline ci/common/suite.sh]: `-r` doesn't work on the macOS envs

**Comment by justinmk** [comment]
BTW is it intentional that tracing is always on? It creates a lot of output...

File: ci/common/suite.sh (hunk 2 above)
Comment (jamessan) [inline ci/common/suite.sh]: You'll need to use `s/[^A-Za-z0-9][^A-Za-z0-9]*/-/g` for the first sed command.  The other one works fine without `-r`.

**Comment by ZyX-I** [comment]
It was supposed to be on inside a suite, but only if it was enabled by the script. Changed to enable/disable it always (not when enabled by the test script) from suite functions instead.

File: ci/common/suite.sh (hunk 2 above)
Comment (ZyX-I) [inline ci/common/suite.sh]: @jamessan `\+` not supported? It runs fine locally without `-r`: just changing `+` to `\+` in regexp.

Comment (jamessan) [inline ci/common/suite.sh]: Or `[^A-Za-z0-9]\{1,\}`.

File: ci/common/suite.sh [hunk 3]
```diff
@@ -11,14 +11,35 @@ FAIL_SUMMARY=""
 END_MARKER="$BUILD_DIR/.tests_finished"
 FAIL_SUMMARY_FILE="$BUILD_DIR/.test_errors"
 
+ANSI_CLEAR="\033[0K"
+
+travis_fold() {
+  local action="$1"
+  local name="$2"
+  name="$(echo -n "$name" | tr '\n\0' '--' | sed 's/[^A-Za-z0-9]\+/-/g')"
+  name="$(echo -n "$name" | sed -r 's/-$//')"
```
Comment (justinmk) [inline ci/common/suite.sh]: still using `-r`

File: ci/common/suite.sh (hunk 2 above)
Comment (ZyX-I) [inline ci/common/suite.sh]: @jamessan Does this “Or” mean that `\+` not supported? Or this is the third alternative?

Comment (jamessan) [inline ci/common/suite.sh]: BRE syntax doesn't support `\+`.

Apple's [re_format(7)](https://developer.apple.com/legacy/library/documentation/Darwin/Reference/ManPages/man7/re_format.7.html#//apple_ref/doc/man/7/re_format) and GNU's [regex(7)](https://manpages.debian.org/jessie/manpages/regex.7.en.html) both describe BRE in similar wording:

```
Obsolete (``basic'') regular expressions differ in several respects.  `|' is an ordinary character and
there is no equivalent for its functionality.  `+' and `?' are ordinary characters, and their functionality
can be expressed using bounds (`{1,}' or `{0,1}' respectively).  Also note that `x+' in modern
REs is equivalent to `xx*'.  The delimiters for bounds are `\{' and `\}', with `{' and `}' by themselves
ordinary characters.
```

**Comment by justinmk** [comment]
The raw log seems to be much smaller. Thanks!

File: ci/common/suite.sh [hunk 4]
```diff
@@ -11,14 +11,35 @@ FAIL_SUMMARY=""
 END_MARKER="$BUILD_DIR/.tests_finished"
 FAIL_SUMMARY_FILE="$BUILD_DIR/.test_errors"
 
+ANSI_CLEAR="\033[0K"
+
+travis_fold() {
+  local action="$1"
+  local name="$2"
+  name="$(echo -n "$name" | tr '\n\0' '--' | sed 's/[^A-Za-z0-9]\{1,\}/-/g')"
```
Comment (jamessan) [inline ci/common/suite.sh]: It would be better to use `printf` instead of `echo` on these lines -- `printf "$name"`

Comment (ZyX-I) [inline ci/common/suite.sh]: @jamessan It must not be `printf "$name"`, should be `printf '%s' "$name"` if you aim at random text. But I know for sure that `$name` always starts with `/` in our code, so no need to bother.

**Comment by ZyX-I** [comment]
Should be ready now.

**Review by jamessan — APPROVED** [review]

**Pull request merged**